dask_expr/_shuffle.py

```diff
@@ -179,54 +141,71 @@ def _layer(self):
 
     @functools.cached_property
     def _meta(self):
-        # We will drop _partitions later on, so reflect this here
-        return self.frame._meta.drop(columns=["_partitions"], errors="ignore")
+        return self.frame._meta
```

Comment:
A little context for other potential reviewers

I noticed weird optimization behaviors where sometimes the `_partitions` column would not be dropped because the optimizer would think that this column wasn't there, i.e. this meta implementation is _wrong_.

This led us to adding explicit `Projection`s below. Since the `from_abstract_shuffle` is doing a similar thing but a little more sophisticated we had to touch this as well.